submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,434 @@
+                </div>
+            </div>`;
+}
+
+function reformatPhoneNumber(number) {
+    return number
+        .replace(/[^0-9]+/g, "") //leave only numbers in phone number
+        .replace(/.(\d{3})/g, "$1-") // add dashes between groups of digits consisting of 3 numbers
+        .replace(/(^\d{3,3})-(.\d+)/, "+($1)-$2") // get first group of 3 digits and place them inside brackets
+        .replace(/[-]+$/g, ""); //remove extra dash if it appers after the last group of numbers
+}
+
+function getDate(date) {
+    return new Date(date).toLocaleString("en-US", {
+        year: "numeric",
+        month: "long",
+        day: "numeric",
+    });
+}
+
+function getGenderIcon(gender) {
+    return gender === "female"
+        ? '<span class="female">♀</span>'
+        : '<span class="male">♂</span>';
+}
+
+function getResponseErrorMessage(status, statusText) {
+    return `<h2 class='error__code'>${status}: ${statusText}</h2>`;
+}
+
+function setAgeLimits(ageLimits, limitInput, limit) {
+    limitInput.min = ageLimits.min;
+    limitInput.max = ageLimits.max;
+    limitInput.value = ageLimits[limit];
+}
+
+function getAgeLimits(friends) {
+    const sortedFriends = friends.sort((a, b) => a.age - b.age);
+    return {
+        min: sortedFriends[0].age,
+        max: sortedFriends[sortedFriends.length - 1].age,
+    };
+}
+
+function sortCards(friends) {
+    const checkedSortInput = document.querySelector(".list__input:checked");
+    if(checkedSortInput){
+        SELECT_CONTAINER.setAttribute("select-modified", true);
+        sortByType[checkedSortInput.getAttribute("value")](friends);
+    }
+}
+
+function updateSelectText(itemToSelect) {
+    SELECT_CONTAINER.textContent = itemToSelect.textContent;
+    SELECT_CONTAINER.setAttribute("value", itemToSelect.getAttribute("value"));
+}
+
+const sortByType = {
+    namesDescending: function(friends) {
+        friends.sort((a, b) => b.firstName.localeCompare(a.firstName));
+    },
+    namesAscending: function(friends) {
+        friends.sort((a, b) => a.firstName.localeCompare(b.firstName));
+    },
+    ageDescending: function(friends) {
+        friends.sort((a, b) => b.age - a.age);
+    },
+    ageAscending: function(friends) {
+        friends.sort((a, b) => a.age - b.age);
+    },
+}
+
+function findSubstring(string, substring) {
+    return string.toLowerCase().includes(substring.toLowerCase());
+}
+
+function findMatchesWithPropertiesValues(friends, substring) {
+    const ALLOWED_SEARCH_FILTER_PROPERTIES = ["firstName", "lastName", "email", "username", "country"];
+    return friends.filter((friend) => {
+        return ALLOWED_SEARCH_FILTER_PROPERTIES
+            .filter((property) => findSubstring(friend[property], substring)).length;
+    });
+}
+
+function filterByGender(friends) {
+    let checkboxes = Array.from(document.querySelectorAll(".checkbox:checked"));
+    return friends.filter((friend) =>
+        checkboxes.some((gender) => friend.gender === gender.value)
+    );
```

Comment:
Why not find only the gender checkbox? 🤔

Reply:
Actually, the `querySelectorAll(".checkbox:checked")` selects only the gender checkboxes. I decided to choose checkboxes instead of radio input, therefore I don't need one more input for `all genders`. To deal with sorting `some` is needed as it iterates through the friends and checks if the friend matches any of `.checkbox:checked` values. In such a way, I am not limited to returning only one specific gender at the time as it sorts by multiple chosen criterias. 
Should I change something?

Reply:
Ah, I see. Ok then!